### DISCUSSION TITLE
Video segment extraction & representation of the segments

### DISCUSSION THREAD

**Discussion by aycatakmaz** [vitrivr/cineast#186] [Ideas]
Dear developers,

I am working on a project as a continuation of the project discussed in the following thread: https://github.com/vitrivr/vitrivr-ng/issues/44.

We are wondering whether it is possible to modify the shot segmentation method. We would like to replace the shot segmentation pipeline with our model, which parts of the code should we modify?

Another question we have is whether there is a possibility of having the video segments played separately on their own, and not as a part of the original, longer-duration video. For instance, although every separate shot from a video is shown as a separate entry in Vitrivr, when we play the video, we see that it is represented as a part of the original video, and it continues playing even after the end of that particular shot. We are wondering if it is possible to make a modification regarding that, such that each shot would be represented as a separate entity.

Thank you very much for your time and help!
 
Best,
Ayça

**Comment by lucaro** [comment]
Dear Ayça,

cineast already has several segmenters (see https://github.com/vitrivr/cineast/tree/master/cineast-core/src/main/java/org/vitrivr/cineast/core/extraction/segmenter) and their use is configurable, so you can just add your own if you need a different functionality. As for the playback, it's probably easiest to extend the UI in order to stop the playback when it reaches the end of a segment. The player already knows about the segment boundaries, so adding this should be relatively simple. Alternatively, you could export the individual segments as video files, but that's probably more trouble than it's worth.

Since this isn't really an issue but rather a question, I'll close this here. If you have any further questions, we'd be happy to continue the discussion over in the [discussion section](https://github.com/vitrivr/cineast/discussions).

Best regards,
Luca

**Comment by silvanheller** [comment]
@lucaro has already answered the segmenter question, i'll just add to the playback part: The easiest way would be to replace the `AdvancedMediaPlayer` in `quick-viewer-component.html` with something that only shows selected parts of the video. A quick look shows that you'll probably want something like [Specifying Playback Range](https://developer.mozilla.org/en-US/docs/Web/Guide/Audio_and_video_delivery#specifying_playback_range)

Btw: I've converted this to a discussion
